frontend/src/metabase/parameters/components/ParameterValueWidget.jsx

```diff
@@ -294,13 +295,29 @@ Widget.propTypes = {
 function getWidgetDefinition(parameter) {
   if (DATE_WIDGETS[parameter.type]) {
     return DATE_WIDGETS[parameter.type];
-  } else if (parameter.hasVariableTemplateTagTarget) {
+  } else if (isTextWidget(parameter)) {
     return TextWidget;
   } else if (isNumberParameter(parameter)) {
     return NumberInputWidget;
-  } else if (!_.isEmpty(parameter.fields)) {
+  } else if (isFieldWidget(parameter)) {
     return ParameterFieldWidget;
   } else {
     return StringInputWidget;
   }
 }
+
+function isTextWidget(parameter) {
```

Comment:
Here we make text tags that have the source specified render with `ParameterFieldWidget`